dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -612,6 +597,52 @@ public ListenableFuture<LoadFileResponse> load(List<LoadSubTask> subTasks, boole
         GrpcExecutors.READER_EXECUTOR);
   }
 
+  private ListenableFuture<Void> submitLoadDataSubTask(
+      Block block, UfsReadOptions options, List<LoadFailure> errors) {
+    ListenableFuture<Void> future =
+        Futures.submit(() -> {
+          try {
+            if (options.hasUser()) {
+              AuthenticatedClientUser.set(options.getUser());
+            }
+            long fileLength = block.getUfsStatus().getUfsFileStatus().getContentLength();
+            if (block.hasMainWorker()) {
+              WorkerNetAddress address = GrpcUtils.fromProto(block.getMainWorker());
```

Comment:
main worker is set from scheduler? if we r submitting load tasks to all replicas at same time, how to ensure the main worker would have loaded at the time of secondary worker trying to read from it?

Reply:
Right now we don't due to the concurrency issue of getAndLoad. So we would recommend to user to load one replica first and then set multiple replicas to ensure they only read the data from ufs once. If we want to further improve this part we can improve the getAndLoad in CacheManager.

Reply:
understood, getAndLoad to save only one read is out of scope on this PR, this PR targets at only save duplicate read from multiple workers not duplicate read from one worker.